File: 4057-total-waviness-of-numbers-in-range-i/total-waviness-of-numbers-in-range-i.c
int waviness(int num) {
    int count = 0;
    if (num < 100) return count;
    int a = num % 10;
    num /= 10;
    int b = num % 10;
    int c = (num / 10) % 10;
    if (a < b && b > c || a > b && b < c) count++;
    return count + waviness(num);
}

int totalWaviness(int num1, int num2) {
    int out = 0;
    for (int n = num1; n <= num2; n++) out += waviness(n);
    return out;
}

File: 4057-total-waviness-of-numbers-in-range-i/total-waviness-of-numbers-in-range-i.py
class Solution:
    # Goober questions goober solutions
    def totalWaviness(self, num1: int, num2: int) -> int:
        return sum(list(map(waveiness := lambda n:0 if n<100 else (((a:=n%10)+(b:=(n//10)%10)+(c:=(n//100)%10))and(a<b and b>c or a>b and b<c))+waveiness(n//10), range(num1, num2+1))))